### DISCUSSION TITLE
view source code format

### DISCUSSION THREAD

**Discussion by ingmardor** [dbeaver/dbeaver#10893] [Q&A] [answered]
Hi!

I am used to use Oracle SQL, but new to PostgreSQL. In Oracle SQL, I can create a VIEW and look into the code I entered later on and the code displayed is the code I entered. But using PostgreSQL via DBeaver, I found the source code of a VIEW I defined looks completely different from what I entered in the editor window. Formatting is changed, keywords are show upper case and text constant I used in the SQL code are now explicitly casted into ::text.

I found I can get a nicer looking code by using the format SQL option. But the result is still different from what I entered.
Is there any way to get the system to store my source code exactly the way I typed it?

Thx for your hints / comments!

**Comment by LonwoLonwo** [comment, marked as answer]
Hello.
We read the DDL of the views directly from the databases.

For Oracle we use this query: "SELECT DBMS_METADATA.GET_DDL('VIEW','VIEWNAME','SCHEMANAME') TXT FROM DUAL"
And for PostgreSQL: "SELECT pg_get_viewdef(viewId, true)"

So unfortunately we cannot change the formatting.
